x-pack/plugins/apm/server/routes/fleet/api_keys/create_apm_server_api_key.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CoreStart, Logger } from '@kbn/core/server';
+
+export async function createApmServerApiKey({
+  coreStart,
+  logger,
+}: {
+  coreStart: CoreStart;
+  logger: Logger;
+}) {
+  logger.debug('Creating APM Server API Key');
+
+  const response =
+    await coreStart.elasticsearch.client.asInternalUser.security.createApiKey({
+      body: {
+        name: 'apm-system-indicies-api-key',
+        metadata: {
+          application: 'apm',
+          consumer: 'apm-server',
+        },
+        role_descriptors: {
+          apmSystemIndices: {
+            cluster: ['all'],
```

Comment:
Are the cluster privileges needed?

Reply:
I'm honestly not sure. @simitt or @kruskall do you know what cluster privileges APM Server needs?

Reply:
Hey @sqren,
apm-server does need cluster privileges:
- `manage_ilm` for the profiling code: https://github.com/elastic/apm-server/blob/main/x-pack/apm-server/profiling/ilm.go
- `manage_api_key`/`manage_own_api_key` for the apiKey command (currently deprecated): https://www.elastic.co/guide/en/apm/guide/current/command-line-options.html#apikey-command
- `manage_api_key`/`manage_own_api_key` for apiKey agent auth: https://www.elastic.co/guide/en/apm/guide/current/api-key.html

Looking at the code the only other usage of the ES client is for querying the cluster UUID (https://github.com/elastic/apm-server/blob/3bf78d92064b241b4b94a0dbe04b2d36e5215f0e/internal/beater/beater.go#L888), unfortunately I do not know if that needs specific cluster permissions.

Reply:
The server shouldn't require any of these cluster privileges for the sourcemap or agent config operations. Let's remove them and test without them.

Reply:
I still see API key created with 3 cluster privileges in the PR. I tested agent config API key without any cluster privileges and it works, so I don't think they are needed

Reply:
@carsonip the cluster privileges have now been removed